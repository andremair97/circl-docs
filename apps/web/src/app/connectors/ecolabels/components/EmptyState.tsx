export default function EmptyState({ label }: { label: string }) {
  return <div className="text-sm text-gray-600 border rounded-xl p-4">No {label} results yet. Try another search.</div>;
}
