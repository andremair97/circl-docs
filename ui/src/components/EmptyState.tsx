import { ReactNode } from 'react';

interface EmptyStateProps {
  icon?: ReactNode;
  title: string;
  description?: string;
  action?: ReactNode;
}

// EmptyState conveys absence of data with optional icon and action.
export default function EmptyState({ icon, title, description, action }: EmptyStateProps) {
  return (
    <div className="py-8 text-center" data-testid="empty-state">
      {icon && <div className="mx-auto mb-4 h-8 w-8 text-primary">{icon}</div>}
      <h2 className="text-lg font-semibold">{title}</h2>
      {description && <p className="mt-2 text-sm text-gray-600">{description}</p>}
      {action && <div className="mt-4">{action}</div>}
    </div>
  );
}
