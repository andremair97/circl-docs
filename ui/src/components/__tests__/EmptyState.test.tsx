import { render, screen } from '@testing-library/react';
import { Search } from 'lucide-react';
import EmptyState from '../EmptyState';

// Snapshot and rendering test for EmptyState component.
it('renders icon, text, and action', () => {
  const { container } = render(
    <EmptyState
      icon={<Search data-testid="icon" />}
      title="No items"
      description="nothing to see"
      action={<button>act</button>}
    />
  );
  expect(screen.getByTestId('icon')).toBeInTheDocument();
  expect(screen.getByText('No items')).toBeInTheDocument();
  expect(screen.getByText('nothing to see')).toBeInTheDocument();
  expect(screen.getByText('act')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
