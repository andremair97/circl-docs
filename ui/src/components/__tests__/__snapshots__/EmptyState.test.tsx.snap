// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders icon, text, and action 1`] = `
<div>
  <div
    class="py-8 text-center"
    data-testid="empty-state"
  >
    <div
      class="mx-auto mb-4 h-8 w-8 text-primary"
    >
      <svg
        class="lucide lucide-search "
        data-testid="icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <path
          d="m21 21-4.3-4.3"
        />
      </svg>
    </div>
    <h2
      class="text-lg font-semibold"
    >
      No items
    </h2>
    <p
      class="mt-2 text-sm text-gray-600"
    >
      nothing to see
    </p>
    <div
      class="mt-4"
    >
      <button>
        act
      </button>
    </div>
  </div>
</div>
`;
